internal/mongodoc/resource.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package mongodoc // import "gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+
+import (
+	"sort"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charm.v6-unstable/resource"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// NewResourceQuery formats the provided details into a mongo query for
+// the identified resource.
+func NewResourceQuery(cURL *charm.URL, resName string, revision int) bson.D {
```

Comment:
We're pretty consistent on the spelling throughout the various code bases.  So I'd rather stick with the shortened name.  Otherwise I agree.